stan/math/prim/mat/fun/trace_inv_quad_form_ldlt.hpp

```diff
@@ -18,10 +18,8 @@ namespace math {
  *       trace(B^T A^-1 B)
  * where the LDLT_factor of A is provided.
  */
-template <typename T1, typename T2, int R2, int C2, int R3, int C3>
-inline typename std::enable_if<
-    !stan::is_var<T1>::value && !stan::is_var<T2>::value,
-    typename boost::math::tools::promote_args<T1, T2>::type>::type
+template <typename T1, typename T2, int R2, int C2, int R3, int C3, typename = enable_if_all_not_var<T1, T2>>
+inline return_type_t<T1, T2>
```

Comment:
This code could still go off for `vari` tho'. Maybe we should have an alias for `enable_if_var_like` etc.